dotnet/src/SemanticKernel.Abstractions/IKernel.cs

```diff
@@ -87,6 +87,16 @@ public interface IKernel
     /// <param name="memory">Semantic memory instance</param>
     void RegisterMemory(ISemanticTextMemory memory);
 
+    /// <summary>
+    /// Run a single synchronous or asynchronous <see cref="ISKFunction"/>.
+    /// </summary>
+    /// <param name="skFunction">A Semantic Kernel function to run</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Result of the function composition</returns>
+    Task<SKContext> RunAsync(
+        ISKFunction skFunction,
+        CancellationToken cancellationToken); // TODO: add `= default` value after the `params` overloads are removed, or it will be ambiguous.
```

Comment:
What's an example where it's ambiguous?  I'd expect this overload to always win for a single ISKFunction argument, even with the default argument.

Reply:
I believe one below with `**params** ISKFunction[]` will result in ambiguity if you only pass
   `k.RunAsync(oneFunction)`

No?

Reply:
I wouldn't expect it to. params should lose. Do you see otherwise?

Reply:
Yup, you're right. Updated